server/lib/devtoolsproxy/proxy.go

```diff
@@ -84,7 +87,37 @@ func (u *UpstreamManager) setCurrent(url string) {
 	if url != "" && url != prev {
 		u.logger.Info("devtools upstream updated", slog.String("url", url))
 		u.currentURL.Store(url)
+		// Broadcast update to subscribers without blocking
+		u.subsMu.RLock()
+		for ch := range u.subs {
+			select {
+			case ch <- url:
+			default:
+			}
+		}
```

Comment:
I can see why we don't want to block necessarily but this implementation _can_ drop messages if the `ch` isn't available to receive...what delivery guarantees should we set?